app/models/manageiq/providers/ibm_cloud/inventory/collector/vpc.rb

```diff
@@ -3,63 +3,108 @@ class ManageIQ::Providers::IbmCloud::Inventory::Collector::VPC < ManageIQ::Provi
   require_nested :NetworkManager
   require_nested :StorageManager
 
-  def connection
-    @connection ||= manager.connect
+  def collect
   end
 
   def vms
-    connection.instances.all
+    all("instances")
   end
 
   def vm_key_pairs(vm_id)
-    connection.instances.instance(vm_id)&.initialization || {}
+    vpc.get_instance_initialization(:id => vm_id)&.result&.dig("keys") || []
   end
 
   def flavors
-    connection.instance_profiles.all
+    vpc.list_instance_profiles&.result&.dig("profiles")
   end
 
   def images
-    @images ||= connection.images.all.to_a
+    @images ||= images_by_id.values
   end
 
   def images_by_id
-    @images_by_id ||= images.index_by { |img| img[:id] }
+    @images_by_id ||= all("images").index_by { |image| image["id"] }
   end
 
   def image(image_id)
-    connection.images.instance(image_id)&.details
+    vpc.get_image(:id => image_id)
   end
 
   def keys
-    connection.keys.all
+    vpc.list_keys&.result&.dig("keys")
   end
 
   def availability_zones
-    connection.regions.instance(manager.provider_region).zones.all
+    vpc.list_region_zones(:region_name => manager.provider_region)&.result&.dig("zones")
   end
 
   def security_groups
-    connection.security_groups.all
+    all("security_groups")
   end
 
   def cloud_networks
-    connection.vpcs.all
+    all("vpcs")
   end
 
   def cloud_subnets
-    connection.subnets.all
+    all("subnets")
   end
 
   def floating_ips
-    connection.floating_ips.all
+    all("floating_ips")
+  end
+
+  def tags(attached_to)
+    global_tagging.list_tags(:attached_to => attached_to)&.result&.dig("items") || []
   end
 
   def volumes
-    connection.volumes.all
+    all("volumes")
   end
 
   def volume(volume_id)
-    connection.volumes.instance(volume_id)&.details
+    vpc.get_volume(:id => volume_id)&.result
+  end
+
+  private
+
+  def api_key
+    @api_key ||= manager.authentication_key
+  end
+
+  def token
+    @token ||= manager.class.raw_connect(api_key)
+  end
+
+  def vpc
+    @vpc ||= begin
+      require "ibm_vpc"
+      service_url = "https://#{manager.provider_region}.iaas.cloud.ibm.com/v1"
```

Comment:
I feel like it would be safer to use a URL generator here, just in case?

```ruby
URI::HTTPS.build(:host => "#{manager.provider_region}.iaas.cloud.ibm.com", :path => "/v1")
```

Reply:
:+1: